### DISCUSSION TITLE
Why are there no standalone records

### DISCUSSION THREAD

**Discussion by tcoopman** [gleam-lang/gleam#3223] [General]
I'm just wondering why this is not valid:

```gleam
State(foo: Int, bar: String)
```

Instead I have to wrap the record in a type: 
```gleam
type State {
  State(foo: Int, bar: String)
}
```

Can someone help me understand this design choice?

**Comment by lpil** [comment]
Gleam is a statically typed language, so all values have to belong to a type. Here you're saying the constructor `State` produces values that belong to the type `State`.

**Reply by tcoopman** [reply]
Thanks that makes sense.

Thinking a bit more about this I wonder if it would make sense to have some syntactic sugar for this. Because I think I often want a type with only one constructor.

I'm not sure this makes sense and if it fits in gleams philosophy but what about this:

```gleam
type State {
  foo: Int, bar: String 
}
```

Like this would be syntactic sugar for:

```gleam
type State {
  State(foo: Int, bar: String)
}
```

Now that I wrote it, I'm not even sure it's a good idea anymore 😄.
Well it's just a brain dump and I'm probably not the first who thought about this, but I couldn't find anything about it.

**Comment by sporto** [comment]
I would like to see records / structs in Gleam instead of having to use custom types all the time.

Couple of reasons. 
Many times you just one a simple record with no variants. So a smaller syntax would be nice, e.g.

```
struct State(foo: Int, bar: String)
```

At the moment, when Gleam generates typescript types it translates custom types to classes. 

```typescript
class State extends _.CustomType {
  constructor(foo: number, bar: string);
  
  foo: number;
  bar: string;
}
```

Which feels unnecessary for a thing that only has one variant (maybe there is a good reason for having this with one variant?). 
It could be nice if a record / struct would translate to a simple TS type like:

```typescript
type State = { state: number, bar: string }
```
